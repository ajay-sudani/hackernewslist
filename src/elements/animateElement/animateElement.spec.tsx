import React from "react";
import renderer from "react-test-renderer";
import Loader from "../loader";
import AnimateElement from "./animateElement";

describe("AnimateElement", () => {
  it("should render with the text as a child prop", () => {
    const animateElement = renderer
      .create(<AnimateElement children="Animated element" />)
      .toJSON();
    expect(animateElement).toMatchSnapshot();
  });

  it("should render with other components as a child prop", () => {
    const animateElement = renderer
      .create(<AnimateElement children={<Loader />} />)
      .toJSON();
    expect(animateElement).toMatchSnapshot();
  });
});
