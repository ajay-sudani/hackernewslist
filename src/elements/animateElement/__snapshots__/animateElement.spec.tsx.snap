// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimateElement should render with other components as a child prop 1`] = `
<div>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(2rem) translateZ(0)",
      }
    }
  >
    <div>
      <span>
        loading
      </span>
      <span />
    </div>
  </div>
</div>
`;

exports[`AnimateElement should render with the text as a child prop 1`] = `
<div>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(2rem) translateZ(0)",
      }
    }
  >
    Animated element
  </div>
</div>
`;
